### PR TITLE
brew.sh: remove annoying prompts when homebrew/core not tapped

### PR DESCRIPTION
**Pull request by XuehaiPan**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [X] Have you successfully run `brew style` with your changes locally?
- [X] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

In the installer ([Homebrew/install](https://github.com/Homebrew/install)), we only officially support three default prefixes (`/usr/local`, `/opt/homebrew` and `/home/linuxbrew/.linuxbrew`). The prefix `~/.linuxbrew` is only available in interactive mode. Discussed in Homebrew/install#582 and Homebrew/install#583, we would not support the non-default prefixes in our official installer (including `~/.linuxbrew` https://github.com/Homebrew/install/pull/583#issuecomment-927826916).

For users who wants non-default prefixes or don't have `sudo` privillages, they should use the [Untar-anywhere](https://docs.brew.sh/Installation#untar-anywhere) / [Alternative-installation](https://docs.brew.sh/Homebrew-on-Linux#alternative-installation) methed (https://github.com/Homebrew/install/pull/583#discussion_r716672112).

```console
$ git clone https://github.com/Homebrew/brew ~/.linuxbrew/Homebrew
Cloning into '/home/PanXuehai/.linuxbrew/Homebrew'...
...

$ mkdir ~/.linuxbrew/bin
$ ln -s ~/.linuxbrew/Homebrew/bin/brew ~/.linuxbrew/bin
$ eval "$(~/.linuxbrew/bin/brew shellenv)"

$ brew update
fatal: cannot change to '/home/PanXuehai/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core': No such file or directory
Already up-to-date.

$ brew update --force --quiet  
fatal: cannot change to '/home/PanXuehai/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core': No such file or directory
fatal: cannot change to '/home/PanXuehai/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core': No such file or directory
fatal: cannot change to '/home/PanXuehai/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core': No such file or directory
HOMEBREW_CORE_GIT_REMOTE set: using https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/linuxbrew-core.git for Homebrew/core Git remote URL.
==> Tapping homebrew/core
Cloning into '/home/PanXuehai/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core'...
...
Already up-to-date.
```

This PR removes the annoying error messages above on Linux and updates the document.

### CODE REVIEW

**Comment by XuehaiPan** [comment]
Cause we only support prefixes (`/usr/local`, `/opt/homebrew` and `/home/linuxbrew/.linuxbrew`) in the official installer non-interactively. And these three prefixes require `sudo` privileges and those requirements are not always met (e.g. non-admin users on a server or cluster). Some "advance" users may want to write automation scripts to set up their shell environment. I wonder if we can document the [Alternative-installation](https://docs.brew.sh/Homebrew-on-Linux#alternative-installation) in a more detailed way:

```bash
HOMEBREW_INSTALL_PREFIX="${HOME}/.linuxbrew"  # pick another prefix at your peril

# Uncomment the following commands to use geolocalized Git mirrors
# export HOMEBREW_BREW_GIT_REMOTE="..."  # put your Git mirror of Homebrew/brew here
# export HOMEBREW_CORE_GIT_REMOTE="..."  # put your Git mirror of Homebrew/homebrew-core here

git clone "${HOMEBREW_BREW_GIT_REMOTE:-https://github.com/Homebrew/brew}" "${HOMEBREW_INSTALL_PREFIX}/Homebrew"
mkdir "${HOMEBREW_INSTALL_PREFIX}/bin"
ln -sfn "../Homebrew/bin/brew" "${HOMEBREW_INSTALL_PREFIX}/bin"
eval "$(${HOMEBREW_INSTALL_PREFIX}/bin/brew shellenv)"
brew update --force --quiet
chmod -R go-w "$(brew --prefix)/share/zsh"
```

File: docs/Homebrew-on-Linux.md [hunk 1]
```diff
@@ -85,8 +85,10 @@ Extract or `git clone` Homebrew wherever you want. Use `/home/linuxbrew/.linuxbr
 ```sh
```
Comment (MikeMcQuaid) [inline docs/Homebrew-on-Linux.md:85]: Let's just remove the entire `Alternative Installation` section; it's linked from the homepage `Install` notes anyway and exists mainly from when Linuxbrew had its own Homebrew/brew fork.

File: docs/Installation.md [hunk 2]
```diff
@@ -51,6 +51,18 @@ here. *Pick another prefix at your peril!*
 
 ```sh
 mkdir homebrew && curl -L https://github.com/Homebrew/brew/tarball/master | tar xz --strip 1 -C homebrew
+eval "$(homebrew/bin/brew shellenv)"
+brew update --force --quiet
+chmod -R go-w "$(brew --prefix)/share/zsh"
```
Comment (MikeMcQuaid) [inline docs/Installation.md:56]:
```suggestion
```

File: docs/Installation.md [hunk 3]
```diff
@@ -51,6 +51,18 @@ here. *Pick another prefix at your peril!*
...
+```
+
+or
+
+```sh
+git clone https://github.com/Homebrew/brew homebrew
+eval "$(homebrew/bin/brew shellenv)"
+brew update --force --quiet
+chmod -R go-w "$(brew --prefix)/share/zsh"
 ```
```
Comment (MikeMcQuaid) [inline docs/Installation.md:66]:
```suggestion
```sh
git clone https://github.com/Homebrew/brew homebrew
#```

then 

```sh
eval "$(homebrew/bin/brew shellenv)"
brew update --force --quiet
chmod -R go-w "$(brew --prefix)/share/zsh"
#```
```

but remove the `#` before the trailing triple backticks

**Review by MikeMcQuaid — COMMENTED** [review]
Looking good so far!

**Review by MikeMcQuaid — APPROVED** [review]
Great work, thanks again @XuehaiPan!

**Pull request merged**